src/garage/sampler/on_policy_vectorized_sampler.py

```diff
@@ -103,27 +103,31 @@ def obtain_samples(self, itr, batch_size=None, whole_paths=True):
 
         policy = self.algo.policy
 
+        agent_infos = policy.get_initial_state(dones)
         while n_samples < batch_size:
             t = time.time()
-            policy.reset(dones)
+            agent_infos = policy.reset(agent_infos, dones)
 
-            actions, agent_infos = policy.get_actions(obses)
+            actions, agent_infos = policy.get_actions(obses, agent_infos)
```

Comment:
please always call keyword arguments as keywords, i.e. `policy.get_actions(obses, agent_infos=agent_infos)` -- this helps avoid really hard to find refactoring bugs.

Reply:
fixed